src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -74,6 +76,19 @@ protected Workspace(HostServices host, string? workspaceKind)
 
             // initialize with empty solution
             _latestSolution = CreateSolution(SolutionId.CreateNewId());
+
+            _taskQueue.ScheduleTask(() => TryRegisterDocumentOptionsProvider(), nameof(TryRegisterDocumentOptionsProvider));
```

Comment:
1. why on a task?  are there race conditions here?
2. what general value is this change trying to provide?  What's wrong with MEF here?

Reply:
Talked to @jasonmalinowski offline and he helped me get to the root of the issue. Will update in a bit. Thanks for taking a look!

Reply:
Sounds good.  Thanks!